api/controllers/MerchStoreController.ts

```diff
@@ -181,15 +187,32 @@ export class MerchStoreController {
     return { error: null, order };
   }
 
-  @Patch('/order')
-  async editMerchOrder(@Body() fulfillOrderRequest: FulfillMerchOrderRequest,
+  @Patch('/order/:uuid')
+  async editMerchOrder(@Params() params: UuidParam,
+    @Body() editOrderRequest: EditMerchOrderRequest,
     @AuthenticatedUser() user: UserModel): Promise<EditMerchOrderResponse> {
+    if (!PermissionsService.canAccessMerchStore(user)) throw new ForbiddenError();
+    const { status, pickupEvent } = editOrderRequest.order;
+    // status and pickup event are not allowed to be updated at the same time,
+    // since they are not logically related, and thus should never have to be updated at the same request
+    if (status && pickupEvent) throw new UserError('Only status or pickupEvent can be updated at once, not both');
+    // members are only allowed to cancel orders. store admins can perform any operation
+    if (status && status !== OrderStatus.CANCELLED && !PermissionsService.canEditMerchStore(user)) {
+      throw new ForbiddenError('Members can only cancel orders');
+    }
+    await this.merchStoreService.editOrder(params.uuid, editOrderRequest.order, user);
+    return { error: null };
+  }
+
+  @Post('/order/:uuid/fulfillment')
+  async fulfillMerchOrder(@Params() params: UuidParam, @Body() fulfillOrderRequest: FulfillMerchOrderRequest,
+    @AuthenticatedUser() user: UserModel): Promise<FulfillMerchOrderResponse> {
     if (!PermissionsService.canFulfillMerchOrders(user)) throw new ForbiddenError();
```

Comment:
I'm hesitant to add hardcoded roles into the backend now because there is an incoming roles rewrite where the ACM admins can create custom roles and define permission levels for each, so for now it might be alright to just have `ADMIN` roles handle this. 

@steets250 do we know have an estimate on wen the roles rewrite might take place? If it's a few months out, we might actually want to define one or more roles for store so that we don't have multiple people on full `ADMIN` accounts.

Reply:
Yeah, actually that's probably a good thing to add before we start adding other new portal features to the backend. I'll work on brainstorming roles + perms this week, and maybe we can further refine at the next work session.